### PR TITLE
Added update channel feature

### PR DESCRIPTION
**Pull request by criticic**
issue: #33 

- Added updateChannel function in workspace controller
- checks:
- - If any field is empty
- - if channel exists
- - if channel created by userId

### CODE REVIEW

File: src/controllers/channel.controller.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    response_201,
+    response_400,
+    response_403,
+    response_500
+} = require('../utils/responseCodes.utils')
+
+
+exports.updateChannel = async (req, res) => {
+    try {
+        const {name, description, userId} = req.body;
```
Comment (automatefreely) [inline src/controllers/channel.controller.js]: userId is not present in request . See issue description.

File: src/controllers/channel.controller.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+        if (!name || !description || !userId || !req.params.id) {
+            return response_400(res, "Invalid Request: Missing required fields")
+        }
+
+        let channel = await Channel.findById(req.params.id);
+        if (!channel) {
```
Comment (automatefreely) [inline src/controllers/channel.controller.js]: You donot have to check if channel exist. Read issue description

**Review by automatefreely — COMMENTED** [review]
Channel and workspace is already checked. Admin is checked already. see the issue description again

File: src/controllers/channel.controller.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    response_400,
+    response_500
+} = require('../utils/responseCodes.utils')
+
+
+exports.updateChannel = async (req, res) => {
+    try {
+        const {name, description} = req.body;
+
+        if (!name || !description || !req.params.id) {
```
Comment (automatefreely) [inline src/controllers/channel.controller.js]: 1. It should be able to update only name or only description or both.
2. In the description of the issue it is mentioned that you are getting the whole channel object. So it better to take the id from that and no need to check for it validity

File: src/controllers/channel.controller.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+            return response_400(res, "Invalid Request: Missing required fields")
+        }
+
+        channel = await Channel.findByIdAndUpdate(req.params.id, {
+            name: name,
+            description: description,
+        });
+
+        return response_201(res, 'Channel Updated Successfully', {
```
Comment (automatefreely) [inline src/controllers/channel.controller.js]: responce code 201 means some thing is created, which is not the case

**Review by automatefreely — COMMENTED** [review]
Make the following changes

File: src/controllers/channel.controller.js (hunk 3 above)
Comment (criticic) [reply src/controllers/channel.controller.js]: I sort of thought the frontend will also send both anyway

**Pull request merged**